### PR TITLE
[libc][mman] Implement msync

### PR DESCRIPTION
**Pull request by SchrodingerZhu**
Implement `msync` as specified in:

1. https://www.man7.org/linux/man-pages/man2/msync.2.html
2. https://pubs.opengroup.org/onlinepubs/9699919799/

### CODE REVIEW

File: libc/test/src/sys/mman/linux/msync_test.cpp [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+TEST(LlvmLibcMsyncTest, LockedPage) {
+  PageHolder page;
+  EXPECT_TRUE(page.is_valid());
+  EXPECT_THAT(LIBC_NAMESPACE::mlock(page.addr, page.size), Succeeds());
+  EXPECT_THAT(
+      LIBC_NAMESPACE::msync(page.addr, page.size, MS_SYNC | MS_INVALIDATE),
+      Fails(EBUSY));
+  EXPECT_THAT(LIBC_NAMESPACE::munlock(page.addr, page.size), Succeeds());
+  EXPECT_THAT(LIBC_NAMESPACE::msync(page.addr, page.size, MS_SYNC), Succeeds());
```
Comment (lntue) [inline libc/test/src/sys/mman/linux/msync_test.cpp:53]: This is a whole sequence of dependent operations, so I think `ASSERT_THAT` should be used instead.

File: libc/test/src/sys/mman/linux/msync_test.cpp [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+}
+
+TEST(LlvmLibcMsyncTest, UnalignedAddress) {
+  PageHolder page;
+  EXPECT_TRUE(page.is_valid());
```
Comment (lntue) [inline libc/test/src/sys/mman/linux/msync_test.cpp:58]: Use `ASSERT_TRUE` to verify the precondtion.

File: libc/test/src/sys/mman/linux/msync_test.cpp [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+  EXPECT_THAT(LIBC_NAMESPACE::msync(&page[1], page.size - 1, MS_SYNC),
+              Fails(EINVAL));
+}
+
+TEST(LlvmLibcMsyncTest, InvalidFlag) {
+  PageHolder page;
+  EXPECT_TRUE(page.is_valid());
```
Comment (lntue) [inline libc/test/src/sys/mman/linux/msync_test.cpp:65]: ditto

**Review by lntue — APPROVED** [review]
LGTM with some nits on the test.

**Comment by SchrodingerZhu** [comment]
nice catch. updated the tests accordingly

**Pull request merged**